features/toolbar/actions.js

```diff
@@ -46,15 +47,19 @@ export function hangup() {
                 .then(() => conference.leave());
         }
 
-        promise = promise
-            .then(() => dispatch(disposeLocalTracks()));
-
         if (connection) {
             promise = promise
                 .then(() => connection.disconnect());
         }
 
-        return promise;
+        // XXX Local tracks and local participant might exist without conference
+        // and connection initialized, so we need to explicitly clean them here.
+        // Furthermore, currently local tracks can be initialized before local
+        // participant is created, so we cannot hope that they will be destroyed
+        // when (and if) local participant leaves.
+        return promise
+            .then(() => dispatch(destroyLocalTracks()))
```

Comment:
We talked about this with @legastero & we agree that in order to continue forward we need to define a sequence of events, ensure that they are guaranteed to fire in this sequence & handle as little as possible in response to each event. @tsareg, could you please start thinking about this sequence without writing source code & let's talk about our thoughts.

For example:
1. There's an event what we associate with creating local tracks. It appears that such an event is very near to the start of the application. Or at the entrance of the welcome page. That's, for example, because @yanas has provided us with UI designs which show the local video on the Welcome page.
2. We define an event that's the opposite of 1. Such an event intuitively has to be near the end of the application. Or at the exit of the Conference page. Hence, the removal of the local tracks from the Redux state & their disposal happens only here i.e. we don't remove local tracks from the Redux state on PARTICIPANT_LEFT.